### ISSUE TITLE
Keep parsing after finding file that cannot be parsed

### ISSUE THREAD

**Issue by xtaixe** [jshiell/checkstyle-idea#527]
Hi.

We have some templates that are not parseable and if we try to scan the whole project we just get "The source file could not be parsed by Checkstyle." and the scan doesn't report anything (I guess it stops).

Ideally:
1. A warning would be shown somewhere every time a file could not be parsed, stating which file.
2. The scan would continue and report for the rest of the files.

**Comment by jshiell** [comment]
I'll have a look when I get a tick, but this is primarily a Checkstyle design contraint, I believe - we pass the list of files to Checkstyle and then parse the events it outputs. The Checkstyle devs have been very explicit in saying that they don't handle unparseable files - this has resuled in a lot of error handling code in the plugin for the inspection scan - and Checkstyle itself has a general habit of not exiting cleanly (often via an Antlr excpetion) when it hits an unparseable file.

So I agree the behaviour above would be preferrable; alas, I suspect we're stuck with Checkstyle blowing up.

Have you considered adding these files to a suppressions file so they are not scanned?

**Comment by xtaixe** [comment]
If this is not related to the plugin, feel free to close the issue.

> Have you considered adding these files to a suppressions file so they are not scanned?

Might try again, but I'm pretty sure I tried that and it wasn't working (or I didn't manage to set it up properly and get it working)

**Comment by xtaixe** [comment]
Actually I already have this:
```
    <!-- Excludes module archetype templates                -->
    <module name="BeforeExecutionExclusionFileFilter">
        <property name="fileNamePattern" value="[/\\]dev[/\\]module\-archetype[/\\]*\.java$"/>
    </module>
```

So unless I'm missing something, this works with Checkstyle (used throught maven plugin) but not with the IntelliJ plugin?

**Comment by xtaixe** [comment]
I don't know how I got there or why Checkstyle is not failing or complaining, but I just realized that expression is wrong. I fixed it seems everything is working both with checkstyle and the IntelliJ plugin, so @jshiell sorry to bother you and feel free to close the issue.
